lib/credentials/basicAuthenticationCredentials.ts

```diff
@@ -44,4 +45,17 @@ export class BasicAuthenticationCredentials implements ServiceClientCredentials
     webResource.headers[HeaderConstants.AUTHORIZATION] = encodedCredentials;
     return Promise.resolve(webResource);
   }
+
+  /**
+   * Signs a request with the Authentication header.
+   *
+   * @param httpRequest The request to be signed.
+   * @returns The signed request.
+   */
+  public signHttpRequest(httpRequest: HttpRequest): Promise<HttpRequest> {
+    const credentials = `${this.userName}:${this.password}`;
+    const encodedCredentials = `${this.authorizationScheme} ${Buffer.from(credentials).toString("base64")}`;
```

Comment:
Is it necessary to specify a character encoding for `Buffer.from`?

Reply:
I'm not sure. This was one of the implementations that I copied and pasted. @amarzavery Do you know?

Reply:
I'm not referring to the "base64" argument to toString--I'm referring to buffer.from. The correct translation of a string into essentially a byte array depends on getting the character encoding right.